### PR TITLE
Document upgrades

### PR DESCRIPTION
**Pull request by errordeveloper**
### Description

Adds documentation for upgrades.

### Checklist

- [x] Added/modified documentation as required (such as the `README.md`, and `examples` directory)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -261,11 +261,130 @@ To uncordon a nodegroup, run:
...
+
+##### Updating multiple nodegroups
+
+If you have multiple nodegroups, it's your responsibility to track how each one was configured.
+You can do this by using config files, but if you haven't used it already, you will need to inspect
+your cluster to find out how each nodegroup was configured.
+
+In general terms, you are looking to:
+- review nodegroups you have and which ones can be delete or must be replaced with new version
+- note down configuration of each nodegroup, consider using config file to ease upgrades next time
```
Comment (rndstr) [inline README.md]: is it obvious where to find the configuration for someone upgrading or do we need to give a hand here?

File: README.md [hunk 2]
```diff
@@ -261,11 +261,130 @@ To uncordon a nodegroup, run:
...
+```
+eksctl create nodegroup --cluster=<clusterName>
+```
+
+Delete old nodegroup:
+```
+eksctl delete nodegroup --cluster=<clusterName> --name=<oldNodeGroupName>
+```
+
+##### Updating multiple nodegroups
```
Comment (rndstr) [inline README.md]: do we want to split this section into `Updating multiple nodegroups with config file` and `Updating multiple nodegroups without config file`?

File: README.md [hunk 3]
```diff
@@ -261,11 +261,130 @@ To uncordon a nodegroup, run:
...
+
+Edit config file to add new nodegroups.
+
+If you chose to deleted definition of old nodegoups from the config file, then you can run:
+```
+eksctl create nodegroup --config-file=<path>
+```
+
+Alternatively, you can pass `--only` with a list of the new nodegroup names that you want to
+add (you can use glob expressions, or list specific names - see above for details):
```
Comment (rndstr) [inline README.md]: not sure where `see above` refers to

File: README.md [hunk 4]
```diff
@@ -261,11 +261,130 @@ To uncordon a nodegroup, run:
...
+To delete old nodegroup:
+```
+eksctl create nodegroup --cluster=<clusterName>
+```
+
+If you are using config file, you will need to do the following.
+
+Get the list of old nodegroups:
+```
+old_nodegroups="$(eksctl get ng --cluster=<clusterName> --output=json | jq -r '.[].Name')"
```
Comment (rndstr) [inline README.md]: why would i need to keep track of the old nodegroups when i use the config file to delete the old ones?

File: README.md [hunk 5]
```diff
@@ -261,11 +261,130 @@ To uncordon a nodegroup, run:
...
+```
+
+Edit config file to add new nodegroups.
+
+If you chose to deleted definition of old nodegoups from the config file, then you can run:
+```
+eksctl create nodegroup --config-file=<path>
```
Comment (rndstr) [inline README.md]: I'm not sure I understand this part, will this command delete nodegroups not in the config file?

File: README.md [hunk 6]
```diff
@@ -261,11 +261,130 @@ To uncordon a nodegroup, run:
...
+```
+
+Once you have new nodegroups in place, you can delete old ones:
+```
+for ng in $old_nodegroups ; do eksctl delete nodegroup --cluster=<clusterName> --name=$ng ; done
+```
+
+#### Updating default add-ons
+
+There 3 default add-ons that get included in each EKS cluster, the process for updating each of them is different, hence
```
Comment (rndstr) [inline README.md]:
```suggestion
There are 3 default add-ons that get included in each EKS cluster, the process for updating each of them is different, hence
```

File: README.md [hunk 7]
```diff
@@ -261,11 +261,130 @@ To uncordon a nodegroup, run:
...
+eksctl utils update-kube-proxy
+```
+
+
+To update `kube-system:ds/aws-node`, run:
+```
+eksctl utils update-aws-node
+```
+
+If you have upgraded from 1.10 to 1.11, you will need to replace `kube-system:deployment/kube-dns` with `kube-system:deployment/coredns`.
```
Comment (rndstr) [inline README.md]: what if i'm going from 1.11 to 1.12, how do i upgrade `kube-dns`?

File: README.md [hunk 8]
```diff
@@ -261,11 +261,130 @@ To uncordon a nodegroup, run:
...
+If you are using config file, you will need to do the following.
+
+Get the list of old nodegroups:
+```
+old_nodegroups="$(eksctl get ng --cluster=<clusterName> --output=json | jq -r '.[].Name')"
+```
+
+Edit config file to add new nodegroups.
+
+If you chose to deleted definition of old nodegoups from the config file, then you can run:
```
Comment (rndstr) [inline README.md]:
```suggestion
If you choose to delete the definition of old nodegroups with the config file, then you can run:
```
(assuming here that nodegroups not found in the config file will be automatically deleted)

File: README.md (hunk 1 above)
Comment (errordeveloper) [reply README.md]: Yes, we will need to, but cannot at the moment. We should move on to caching config inside the cluster, perhaps in a configmap to start with. This would need to be solved as we get closer to `eksctl apply` (also see #443).

File: README.md (hunk 2 above)
Comment (errordeveloper) [reply README.md]: Not sure at the moment, can always improve later on. We actually need a proper website for docs somewhat urgently.

File: README.md (hunk 3 above)
Comment (errordeveloper) [reply README.md]: Well spotted, will try to clarify.

File: README.md (hunk 4 above)
Comment (errordeveloper) [reply README.md]: I guess it all depends which way you edit the file and how you apply the edits, so you could delete old ones from the file to begin with avoid having to set `--only`. There is definitely much room for improvement of the UX overall, this one of the somewhat tricky things.

File: README.md (hunk 5 above)
Comment (errordeveloper) [reply README.md]: No, it will only create new ones, and unless you've deleted the definitions of the old ones before hand - it will complain about those. You'd want to set `--only=<listOfNewNGs>` otherwise.

File: README.md (hunk 7 above)
Comment (errordeveloper) [reply README.md]: That's unknown at present.

File: README.md [hunk 9]
```diff
@@ -261,11 +261,130 @@ To uncordon a nodegroup, run:
 eksctl drain nodegroup --cluster=<clusterName> --name=<nodegroupName> --undo
 ```
 
+### Cluster Upgrades
+
+An _`eksctl`-managed_ cluster can be upgraded in 3 easy steps:
+
+1. update control plane version with `eksctl update cluster`
+2. replace each of the nodegroups by creating a new one and deleting the old one
+3. update default add-ons:
```
Comment (errordeveloper) [inline README.md]: There a few objects there, we should just name the add-ons instead of referring to the workload objects...

File: README.md (hunk 5 above)
Comment (errordeveloper) [reply README.md]: I think `--only` is documented quite clearly just below this.

File: README.md [hunk 10]
```diff
@@ -261,11 +261,133 @@ To uncordon a nodegroup, run:
...
+If you are using config file, you will need to do the following.
+
+Get the list of old nodegroups:
+```
+old_nodegroups="$(eksctl get ng --cluster=<clusterName> --output=json | jq -r '.[].Name')"
+```
+
+Edit config file to add new nodegroups.
+
+If you choose to delete the definition of old nodegroups from the config file, then you can run:
```
Comment (rndstr) [inline README.md]: i understood now, i think the `delete` made me think of `eksctl delete`, I would reword as such:
```suggestion
If you have removed the definition of the old nodegroups from the config file, then you can run:
```

File: README.md [hunk 11]
```diff
@@ -261,11 +261,133 @@ To uncordon a nodegroup, run:
...
+```
+eksctl create nodegroup --config-file=<path>
+```
+
+Alternatively, you can keep the old definitions and pass `--only` with a list of the new nodegroup
```
Comment (rndstr) [inline README.md]:
```suggestion
Alternatively, you can keep the old definitions in the config-file and pass `--only` with a list of the new nodegroup
```

**Comment by whereisaaron** [comment]
Let me hear it for doc-u-men-tation! Woop! Woop!
